cpp/include/cudf/rolling.hpp

```diff
@@ -379,6 +381,125 @@ std::unique_ptr<column> grouped_time_range_rolling_window(
+ *    {   "bottas",        2,            2        },
+ *    {   "bottas",        1,            3        },
+ *    {   "lewis",         3,            1        },
+ *    {   "lewis",         8,            2        },
+ *    {   "bottas",        5,            7        },
+ *    {   "bottas",        6,            8        },
+ *    {   "lewis",         4,            4        }
+ *  ]
+ *
+ * Partitioning (grouping) by `driver_name`, and ordering by `lap_number` yields the following
+ * `num_overtakes` vector (with 2 groups, one for each distinct `driver_name`):
+ *
+ * lap_number:      [ 1,  2,  3,  7,  8,   1,  1,   2,  4 ]
+ * num_overtakes:   [ 1,  2,  1,  5,  6,   2,  3,   8,  4 ]
+ *                    <-----bottas------>|<----lewis------>
+ *
+ * The SUM aggregation is applied, with 1 preceding, and 1 following, with a minimum of 1
+ * period. The aggregation window is thus 3 (laps) wide, yielding the following output column:
+ *
+ *  Results:        [ 3,  4,  3,  11, 11,  13, 13,  13,  4 ]
+ *
+ * @endcode
+ *
+ * Note: The number of rows participating in each window might vary, based on the index within the
+ * group, datestamp, and `min_periods`. Apropos:
+ *  1. results[0] considers 2 values, because it is at the beginning of its group, and has no
+ *     preceding values.
+ *  2. results[5] considers 3 values, despite being at the beginning of its group. It must include 2
+ *     following values, based on its orderby_column value.
+ *
+ * Each aggregation operation cannot cross group boundaries.
+ *
+ * The type of the returned column depends on the input column type `T`, and the aggregation:
+ *   1. COUNT   returns `INT32` columns
+ *   2. MIN/MAX returns `T` columns
+ *   3. SUM     returns the promoted type for T. Sum on `INT32` yields `INT64`.
+ *   4. MEAN    returns FLOAT64 columns
+ *   5. COLLECT returns columns of type `LIST<T>`.
+ *
+ * LEAD/LAG/ROW_NUMBER are undefined for range queries.
+ *
+ * @param[in] group_keys The (pre-sorted) grouping columns
+ * @param[in] orderby_column The (pre-sorted) order-by column, for range comparisons
+ * @param[in] order  The order (ASCENDING/DESCENDING) in which the order-by column is sorted
+ * @param[in] input The input column (to be aggregated)
+ * @param[in] preceding The interval value in the backward direction
+ * @param[in] following The interval value in the forward direction.
+ * @param[in] min_periods Minimum number of observations in window required to have a value,
+ *                        otherwise element `i` is null.
+ * @param[in] aggr The rolling window aggregation type (SUM, MAX, MIN, etc.)
+ *
+ * @returns   A nullable output column containing the rolling window results
+ */
+std::unique_ptr<column> grouped_range_rolling_window(
+  table_view const& group_keys,
+  column_view const& orderby_column,
+  cudf::order const& order,
+  column_view const& input,
+  range_window_bounds&& preceding,
+  range_window_bounds&& following,
```

Comment:
Why are these r-value refs? What if someone wants to reuse the same `range_window_bounds` for multiple calls to `grouped_range_rolling_window`?

Reply:
This had to do with the ownership of the range scalar in `range_window_bounds`. I'll see if this can be removed if `scale_to` is removed.